core/src/main/java/org/bitcoinj/base/internal/ByteUtils.java

```diff
@@ -95,78 +100,99 @@ public static BigInteger bytesToBigInteger(byte[] bytes) {
         return new BigInteger(1, bytes);
     }
 
+    /** Write 2 bytes to the buffer as unsigned 16-bit integer in little endian format. */
+    public static void writeUint16LE(int val, ByteBuffer buf) throws BufferOverflowException {
+        buf.order(ByteOrder.LITTLE_ENDIAN).putShort((short) val);
```

Comment:
Do we want to verify that `int val` is not going to lose information when cast to `short`?  (The existing methods don't do this, so not strictly necessary.)

Reply:
Maybe we should. I'm not planning to deprecate the variants that take an array, so I'd say yes.

Reply:
Would it be useful to return the `ByteBuffer` from this method instead of `void`? This might allow some method chaining and/or make conversion to using `ByteBuffer` easier in some use-cases?

Reply:
Actually it would be a better idea to check for the exact bounds, depending on if the methods take signed or unsigned values. I'll prepare something, probably as a followup commit.

Reply:
> Would it be useful to return the `ByteBuffer` from this method instead of `void`? This might allow some method chaining and/or make conversion to using `ByteBuffer` easier in some use-cases?

Great idea! Implemented.